R/estimateDiversity.R

```diff
@@ -313,25 +313,33 @@ setMethod("estimateDiversity", signature = c(x="TreeSummarizedExperiment"),
             
             # Faith will be calculated
             calc_faith <- TRUE
-        } else{
+        } 
+        # throw warning if there's no tree
+        else if (is.null(rowTree(x))) {
+            warning("Faith diversity will not be calculated because there's 
+                    no phylogenetic tree.")
+            calc_faith <- FALSE
+        }
+        
+        else{
             # Faith will not be calculated
             calc_faith <- FALSE
         }
         
         # If index list contained other than 'faith' index, the length of the
         # list is over 0
-        if( length(index)>0){
+        if(length(index)>0){
             # Calculates all indices but not 'faith'
             x <- callNextMethod()
         }
         # If 'faith' was one of the indices, 'calc_faith' is TRUE
-        if( calc_faith ){
+        if(calc_faith ){
             # Get tree to check whether faith can be calculated
             tree <- rowTree(x, tree_name)
             # Check if faith can be calculated. Give warning and do not run estimateFaith
             # if there is no rowTree and other indices were also calculated. Otherwise, 
             # run estimateFaith. (If there is no rowTree --> error)
-            if( (is.null(tree) || is.null(tree$edge.length)) &&
```

Comment:
Same here and below. The functionality is not changed so this it not well-reasoned change